demo/admin/src/products/future/generated/CombinationFieldsTestProductsGrid.tsx

```diff
@@ -317,6 +316,113 @@ export function ProductsGrid({ toolbarAction, rowAction, actionsColumnWidth = 52
             flex: 1,
             minWidth: 150,
         },
+        {
+            field: "nestedGroups",
+            headerName: intl.formatMessage({ id: "product.nestedGroups", defaultMessage: "Nested groups" }),
+            filterable: false,
+            sortable: false,
+            renderCell: ({ row }) => {
+                const secondaryTextNestedItem3GroupValues: string[] = [
```

Comment:
Where does the "3" come from?

Reply:
From here: `nestedItem${(index + 1).toString()}`

https://github.com/vivid-planet/comet/blob/a0ff603d7ca4f2178df5efd896c26b5b1c71e62d/packages/admin/cms-admin/src/generator/future/generateGrid/combinationColumn.ts#L132-L135

This is required to make sure message-IDs and variable-names from items in a group are named unique, as unlike the `formattedMessage` type, the values of a `group` type do not have a unique key (`fields` is an array, instead of an object):

```ts
{
    type: "formattedMessage",
    message: "Product: {title} is of category {category}",
    valueFields: {
        title: "title",
        category: { type: "text", field: "category.title" },
    },
};
```
```ts
{
    type: "group",
    fields: [
        { type: "text", field: "title" },
        { type: "number", field: "price", currency: "EUR" },
    ],
}
```

### Therefore removing `index` as a unique value would cause the following issues:

Using multiple `type: "static"` values would create two equal message-IDs with different text:

```ts
primaryText: {
    type: "formattedMessage",
    message: "Static values: ({staticValues})",
    valueFields: {
        staticValues: {
            type: "group",
            fields: [
                { type: "static", text: "Example static text" },
                { type: "static", text: "Another static text" },
            ],
        },
    },
}

// results in:
const primaryTextStaticValuesGroupValues: string[] = [
    intl.formatMessage({ id: "product.groupInFormattedMessage.primaryText.staticValues.nestedItem", defaultMessage: `Example static text` }),
    intl.formatMessage({ id: "product.groupInFormattedMessage.primaryText.staticValues.nestedItem", defaultMessage: `Another static text` }),
];
```

And would cause this to create two variables with the same name:

```ts
primaryText: {
    type: "group",
    fields: [
        {
            type: "group",
            fields: [
                { type: "text", field: "title" },
                { type: "text", field: "type" },
            ],
        },
        {
            type: "group",
            fields: [
                { type: "text", field: "category.title" },
                { type: "text", field: "description" },
            ],
        },
    ],
},

// results in 
const primaryTextNestedItemGroupValues: string[] = [row.title ?? "", row.type ?? ""];
const primaryTextNestedItemGroupValues: string[] = [row.category?.title ?? "", row.description ?? ""];
const primaryTextGroupValues: string[] = [
    primaryTextNestedItemGroupValues.filter(Boolean).join(" • "),
    primaryTextNestedItemGroupValues.filter(Boolean).join(" • "),
];
```

---

I know these examples are very niece/unrealistic for a practical use-case, however this makes sure future potential types that require unique values (message-IDs, variables) will work when used inside `type: "group"`.  

An alternative, would be to require a key for each item in `fields` (use an object instead of an array). 
Imo an array is easier to configure though.